python/ray/serve/api.py

```diff
@@ -296,62 +282,44 @@ def deployment(
-            Routing is done based on longest-prefix match, so if you have
-            deployment A with a prefix of '/a' and deployment B with a prefix
-            of '/a/b', requests to '/a', '/a/', and '/a/c' go to A and requests
-            to '/a/b', '/a/b/', and '/a/b/c' go to B. Routes must not end with
-            a '/' unless they're the root (just '/'), which acts as a
-            catch-all.
-        ray_actor_options (Default[Dict]): Options to be passed to the Ray
-            actor constructor such as resource requirements. Valid options are
-            `accelerator_type`, `memory`, `num_cpus`, `num_gpus`,
-            `object_store_memory`, `resources`, and `runtime_env`.
-        user_config (Default[Optional[Any]]): Config to pass to the
-            reconfigure method of the deployment. This can be updated
-            dynamically without changing the version of the deployment and
-            restarting its replicas. The user_config must be json-serializable
-            to keep track of updates, so it must only contain json-serializable
-            types, or json-serializable types nested in lists and dictionaries.
-        max_concurrent_queries (Default[int]): The maximum number of queries
-            that will be sent to a replica of this deployment without receiving
-            a response. Defaults to 100.
-        is_driver_deployment (Optional[bool]): [Experiment] when set it as True, serve
-            will deploy exact one deployment to every node.
+    """Decorator that converts a Python class to a `Deployment`.
 
     Example:
-    >>> from ray import serve
-    >>> @serve.deployment(name="deployment1") # doctest: +SKIP
-    ... class MyDeployment: # doctest: +SKIP
-    ...     pass # doctest: +SKIP
 
-    >>> MyDeployment.bind(*init_args) # doctest: +SKIP
-    >>> MyDeployment.options( # doctest: +SKIP
-    ...     num_replicas=2, init_args=init_args).bind()
+    .. code-block:: python
+
+        from ray import serve
+
+        @serve.deployment(num_replicas=2)
+        class MyDeployment:
+            pass
+
+        app = MyDeployment.bind()
+
+    Args:
+        name: Name uniquely identifying this deployment within the application.
+            If not provided, the name of the class or function will be used.
+        num_replicas: The number of replicas to run that will handle requests to
+            this deployment. Defaults to 1.
+        init_args: [DEPRECATED] These should be passed to `.bind()` instead.
+        init_kwargs: [DEPRECATED] These should be passed to `.bind()` instead.
+        route_prefix: Requests to paths under this HTTP path prefix will be routed
+            to this deployment. Defaults to '/{name}'. This can only be set for the
+            ingress (top-level) deployment of an application.
+        ray_actor_options: Options to be passed to the Ray actor decorator such as
+            resource requirements. Valid options are `accelerator_type`, `memory`,
```

Comment:
I think this has some of the configs left out like autoscaling config, healthcheck, graceful shutdown

Reply:
added!